src/core/middleware.js

```diff
@@ -45,7 +50,19 @@ export function prefixMiddleWare(req, res, next, { _config = config } = {}) {
     // We skip prepending an app if we'd previously prepended a lang and the
     // 2nd part of the URL is now a valid app.
     log.info('Application in URL is valid following prepending a lang.');
-  } else if (!hasValidClientApp) {
+  } else if (
+    (hasValidUrlException || isValidUrlException(URLParts[1], { _config }))
+  ) {
+    if (hasValidLang) {
+      log.info('Exception in URL found; we fallback to addons-server.');
+
+      res.status(404).end(
+        'This page exists only on addons-server; returning 404 to fallback.');
+      return next();
+    }
+
+    log.info('Exception in URL found; prepending lang to URL.');
+  } else if (!hasValidClientApp && !hasValidUrlException) {
```

Comment:
When I remove `&& !hasValidUrlException` from the if statement, all tests pass. Is this one needed? Can you add test coverage?